File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/commercenotificationtemplate/CommercenotificationtemplateService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.commercenotificationtemplate;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class CommercenotificationtemplateService extends BaseService {

	public CommercenotificationtemplateService(Session session) {
		super(session);
	}

	public JSONObject addCommerceNotificationTemplate(long userId, long groupId, String name, String description, String from, JSONObject fromNameMap, String to, String cc, String bcc, String type, boolean enabled, JSONObject subjectMap, JSONObject bodyMap, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("bcc", checkNull(bcc));
			_params.put("bodyMap", checkNull(bodyMap));
			_params.put("cc", checkNull(cc));
			_params.put("description", checkNull(description));
			_params.put("enabled", enabled);
			_params.put("from", checkNull(from));
			_params.put("fromNameMap", checkNull(fromNameMap));
			_params.put("groupId", groupId);
			_params.put("name", checkNull(name));
			_params.put("subjectMap", checkNull(subjectMap));
			_params.put("to", checkNull(to));
			_params.put("type", checkNull(type));
			_params.put("userId", userId);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commercenotificationtemplate/add-commerce-notification-template", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject addCommerceNotificationTemplate(String name, String description, String from, JSONObject fromNameMap, String to, String cc, String bcc, String type, boolean enabled, JSONObject subjectMap, JSONObject bodyMap, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("bcc", checkNull(bcc));
			_params.put("bodyMap", checkNull(bodyMap));
			_params.put("cc", checkNull(cc));
			_params.put("description", checkNull(description));
			_params.put("enabled", enabled);
			_params.put("from", checkNull(from));
			_params.put("fromNameMap", checkNull(fromNameMap));
			_params.put("name", checkNull(name));
			_params.put("subjectMap", checkNull(subjectMap));
			_params.put("to", checkNull(to));
			_params.put("type", checkNull(type));
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commercenotificationtemplate/add-commerce-notification-template", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public void deleteCommerceNotificationTemplate(long commerceNotificationTemplateId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceNotificationTemplateId", commerceNotificationTemplateId);

			_command.put("/commerce.commercenotificationtemplate/delete-commerce-notification-template", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public JSONObject getCommerceNotificationTemplate(long commerceNotificationTemplateId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceNotificationTemplateId", commerceNotificationTemplateId);

			_command.put("/commerce.commercenotificationtemplate/get-commerce-notification-template", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getCommerceNotificationTemplates(long groupId, boolean enabled, int start, int end, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("enabled", enabled);
			_params.put("end", end);
			_params.put("groupId", groupId);
			_params.put("start", start);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator<com.liferay.commerce.notification.model.CommerceNotificationTemplate>", orderByComparator);

			_command.put("/commerce.commercenotificationtemplate/get-commerce-notification-templates", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONArray getCommerceNotificationTemplates(long groupId, int start, int end, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("end", end);
			_params.put("groupId", groupId);
			_params.put("start", start);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator<com.liferay.commerce.notification.model.CommerceNotificationTemplate>", orderByComparator);

			_command.put("/commerce.commercenotificationtemplate/get-commerce-notification-templates", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer getCommerceNotificationTemplatesCount(long groupId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("groupId", groupId);

			_command.put("/commerce.commercenotificationtemplate/get-commerce-notification-templates-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public Integer getCommerceNotificationTemplatesCount(long groupId, boolean enabled) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("enabled", enabled);
			_params.put("groupId", groupId);

			_command.put("/commerce.commercenotificationtemplate/get-commerce-notification-templates-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public JSONObject updateCommerceNotificationTemplate(long commerceNotificationTemplateId, String name, String description, String from, JSONObject fromNameMap, String to, String cc, String bcc, String type, boolean enabled, JSONObject subjectMap, JSONObject bodyMap, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("bcc", checkNull(bcc));
			_params.put("bodyMap", checkNull(bodyMap));
			_params.put("cc", checkNull(cc));
			_params.put("commerceNotificationTemplateId", commerceNotificationTemplateId);
			_params.put("description", checkNull(description));
			_params.put("enabled", enabled);
			_params.put("from", checkNull(from));
			_params.put("fromNameMap", checkNull(fromNameMap));
			_params.put("name", checkNull(name));
			_params.put("subjectMap", checkNull(subjectMap));
			_params.put("to", checkNull(to));
			_params.put("type", checkNull(type));
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commercenotificationtemplate/update-commerce-notification-template", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}

File: windows/Liferay.SDK/Service/V73/Commercenotificationtemplate/CommercenotificationtemplateService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Commercenotificationtemplate
{
	public class CommercenotificationtemplateService : ServiceBase
	{
		public CommercenotificationtemplateService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> GetCommerceNotificationTemplateAsync(long commerceNotificationTemplateId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceNotificationTemplateId", commerceNotificationTemplateId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercenotificationtemplate/get-commerce-notification-template", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddCommerceNotificationTemplateAsync(string name, string description, string from, IDictionary<string, string> fromNameMap, string to, string cc, string bcc, string type, bool enabled, IDictionary<string, string> subjectMap, IDictionary<string, string> bodyMap, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("name", name);
			_parameters.Add("description", description);
			_parameters.Add("from", from);
			_parameters.Add("fromNameMap", fromNameMap);
			_parameters.Add("to", to);
			_parameters.Add("cc", cc);
			_parameters.Add("bcc", bcc);
			_parameters.Add("type", type);
			_parameters.Add("enabled", enabled);
			_parameters.Add("subjectMap", subjectMap);
			_parameters.Add("bodyMap", bodyMap);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commercenotificationtemplate/add-commerce-notification-template", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddCommerceNotificationTemplateAsync(long userId, long groupId, string name, string description, string from, IDictionary<string, string> fromNameMap, string to, string cc, string bcc, string type, bool enabled, IDictionary<string, string> subjectMap, IDictionary<string, string> bodyMap, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("groupId", groupId);
			_parameters.Add("name", name);
			_parameters.Add("description", description);
			_parameters.Add("from", from);
			_parameters.Add("fromNameMap", fromNameMap);
			_parameters.Add("to", to);
			_parameters.Add("cc", cc);
			_parameters.Add("bcc", bcc);
			_parameters.Add("type", type);
			_parameters.Add("enabled", enabled);
			_parameters.Add("subjectMap", subjectMap);
			_parameters.Add("bodyMap", bodyMap);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commercenotificationtemplate/add-commerce-notification-template", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task DeleteCommerceNotificationTemplateAsync(long commerceNotificationTemplateId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceNotificationTemplateId", commerceNotificationTemplateId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercenotificationtemplate/delete-commerce-notification-template", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<IEnumerable<dynamic>> GetCommerceNotificationTemplatesAsync(long groupId, int start, int end, JsonObjectWrapper orderByComparator)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "orderByComparator", "$languageUtil.getJSONWrapperClassName($parameter.type)", orderByComparator);

			var _command = new JsonObject()
			{
				{ "/commerce.commercenotificationtemplate/get-commerce-notification-templates", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceNotificationTemplatesAsync(long groupId, bool enabled, int start, int end, JsonObjectWrapper orderByComparator)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("enabled", enabled);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "orderByComparator", "$languageUtil.getJSONWrapperClassName($parameter.type)", orderByComparator);

			var _command = new JsonObject()
			{
				{ "/commerce.commercenotificationtemplate/get-commerce-notification-templates", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetCommerceNotificationTemplatesCountAsync(long groupId, bool enabled)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("enabled", enabled);

			var _command = new JsonObject()
			{
				{ "/commerce.commercenotificationtemplate/get-commerce-notification-templates-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetCommerceNotificationTemplatesCountAsync(long groupId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercenotificationtemplate/get-commerce-notification-templates-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> UpdateCommerceNotificationTemplateAsync(long commerceNotificationTemplateId, string name, string description, string from, IDictionary<string, string> fromNameMap, string to, string cc, string bcc, string type, bool enabled, IDictionary<string, string> subjectMap, IDictionary<string, string> bodyMap, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceNotificationTemplateId", commerceNotificationTemplateId);
			_parameters.Add("name", name);
			_parameters.Add("description", description);
			_parameters.Add("from", from);
			_parameters.Add("fromNameMap", fromNameMap);
			_parameters.Add("to", to);
			_parameters.Add("cc", cc);
			_parameters.Add("bcc", bcc);
			_parameters.Add("type", type);
			_parameters.Add("enabled", enabled);
			_parameters.Add("subjectMap", subjectMap);
			_parameters.Add("bodyMap", bodyMap);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commercenotificationtemplate/update-commerce-notification-template", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}